src/git.md

```diff
@@ -0,0 +1,187 @@
+ updates. You can use `git status` to check if there are any files you forgot
+ to stage.
+ 6. Push your changes to your fork: `git push -u origin issue-12345-fix`.
+ 7. [Open a PR][ghpullrequest] from your fork to rust-lang/rust's master branch.
+
+[ghpullrequest]: https://guides.github.com/activities/forking/#making-a-pull-request
+
+If your reviewer requests changes, the procedure for those changes looks much
+the same, with some steps skipped:
+
+ 1. Ensure that you're making changes to the most recent version of your code:
+ `git checkout issue-12345-fix`.
+ 2. Make, stage, and commit your additional changes just like before.
+ 3. Push those changes to your fork: `git push`.
+
+## Conflicts
+
+When you edit your code locally, you are making changes to the version of
+rust-lang/rust that existed when you created your feature branch. As such, when
+you submit your PR it is possible that some of the changes that have been made
+to rust-lang/rust since then are in conflict with the changes you've made.
+
+When this happens, you need to resolve the conflicts before your changes can be
+merged. First, get a local copy of the conflicting changes: Checkout your local
+master branch with `git checkout master`, then `git pull rust master` to
+update it with the most recent changes.
+
+### Rebasing
+
+You're now ready to start the rebasing process. Check out the branch with your
+changes and execute `git rebase master`.
+
+When you rebase a branch on master, all the changes on your branch are
+reapplied to the most recent version of master. In other words, git tries to
+pretend that the changes you made to the old version of master were instead
+made to the new version of master. During this process, you should expect to
+encounter at least one "rebase conflict." This happens when git's attempt to
+reapply the changes fails because your changes conflicted with other changes
+that have been made. You can tell that this happened because you'll see
+lines in the output that look like
+
+```
+CONFLICT (content): Merge conflict in file.rs
+```
+
+When you open these files, you'll see sections of the form
+
+```
+<<<<<<< HEAD
+Original code
+=======
+Your code
+>>>>>>> 8fbf656... Commit fixes 12345
+```
+
+This represents the lines in the file that git could not figure out how to
+rebase. The section between `<<<<<<< HEAD` and `=======` has the code from
+master, while the other side has your version of the code. You'll need to
+decide how to deal with the conflict. You may want to keep your changes,
+keep the changes on master, or combine the two.
+ 
+Generally, resovling the conflict consists of two steps: First, fix the
+particular conflict. Edit the file to make the changes you want and remove the
+`<<<<<<<`, `=======` and `>>>>>>>` lines in the process. Second, check the
+surrounding code. If there was a conflict, its because someone else changed the
+same code you did. That means its likely there are some logical errors lying
+around too!
```

Comment:
Maybe we should suggest `x.py check` too? x.py compiles a lot more than it needs to if you're only working on the library though :/ `x.py check library/std` won't do extra work but also won't check enough if you're working on the compiler itself.

```suggestion
around too! It's a good idea to run `x.py check` here to make sure there are no glaring errors.
```